bin/user/mqtt.py

```diff
@@ -537,3 +559,52 @@ def process_record(self, record, dbm):
                 if res != mqtt.MQTT_ERR_SUCCESS:
                     logerr("publish failed for %s: %s" %
                            (tpc, mqtt.error_string(res)))
+
+    PERIODS = {
+        'day':lambda _time_ts:weeutil.weeutil.archiveDaySpan(_time_ts),
+        'yesterday':lambda _time_ts:weeutil.weeutil.archiveDaySpan(_time_ts,1,1),
+        'week':lambda _time_ts:weeutil.weeutil.archiveWeekSpan(_time_ts),
+        'month':lambda _time_ts:weeutil.weeutil.archiveMonthSpan(_time_ts),
+        'year':lambda _time_ts:weeutil.weeutil.archiveYearSpan(_time_ts)}
+
+    def get_record(self, record, dbmanager):
+        """Augment record data with additional data from the archive.
+        Should return results in the same units as the record and the database.
+
+        returns: A dictionary of weather values"""
+
+        # run parent class
+        _datadict = super(MQTTThread,self).get_record(record,dbmanager)
+
+        # actual time stamp
+        _time_ts = _datadict['dateTime']
+
+        # go through all calculations
+        for agg_obs in self.calculations:
+            try:
+                tag = self.calculations[agg_obs].split('.')
+                if len(tag)==3:
```

Comment:
What happens otherwise? When can the length of tag be other than 3? I like to log a warning for this kind of thing.

Reply:
Yes, it makes sense to create some error message.

Reply:
Resolved.